ui/ui-config/pages/profile/new.tsx

```diff
@@ -0,0 +1,85 @@
+import Head from "next/head";
+import { useState } from "react";
+import { useApi } from "@grouparoo/ui-components/hooks/useApi";
+import { useForm } from "react-hook-form";
+import { useRouter } from "next/router";
+import { Form } from "react-bootstrap";
+import LoadingButton from "@grouparoo/ui-components/components/loadingButton";
+import { Actions } from "@grouparoo/ui-components/utils/apiData";
+
+export default function NewProfile(props) {
+  const { errorHandler } = props;
+  const router = useRouter();
+  const { execApi } = useApi(props, errorHandler);
+  const { handleSubmit, register } = useForm();
+  const [loading, setLoading] = useState(false);
+
+  async function onSubmit(data) {
+    setLoading(true);
+    const response: Actions.ProfileCreate = await execApi(
+      "post",
+      `/profile`,
+      Object.assign({}, { properties: { [data.uniqueProperty]: data.value } })
+    );
+    if (response?.profile) {
+      router.push(`/profile/[id]/edit`, `/profile/${response.profile.id}/edit`);
+    } else {
+      setLoading(false);
+    }
+  }
+
+  const uniqueDirectlyMappedProfileProperties = props.properties.filter(
+    (rule) => rule.unique && rule.directlyMapped
+  );
```

Comment:
It might be helpful to give ` props.properties` a type of `Actions.PropertiesList`